doc/man7/EVP_KEM-MLKEM.pod

```diff
@@ -0,0 +1,65 @@
+=pod
+
+=head1 NAME
+
+EVP_KEM-ML-KEM-512,EVP_KEM-ML-KEM-768,EVP_KEM-ML-KEM-1024,EVP_KEM-MLKEM
+- EVP_KEM ML-KEM keytype and algorithm support
+
+=head1 DESCRIPTION
+
+The B<ML-KEM> keytypes and parameters are described in L<EVP_PKEY-MLKEM(7)>.
```

Comment:
Particularly, given the section title immediately below...
```suggestion
The B<ML-KEM> keytypes and parameters are described in L<EVP_PKEY-ML-KEM(7)>.
```

Reply:
We'll probably have to re-do it all as and when there's a final decision on how to name things. And given that, we'll re-visit the documentation as a whole when we're getting closer to merge-to-master -- which we're imo still pretty far away from (PR feedback, encoding/decoding, seeing that hybrids work, etc.)

Reply:
Well...  this algorithm *is* commonly referred to as `ML-KEM` (for example: https://openquantumsafe.org/liboqs/algorithms/kem/ml-kem.html).  I for one don't see why we would need to create another variant of the name (i.e. without the dash).

Reply:
That discussion happens elsewhere. IETF uses no-dashes, users probably would prefer it, but NIST defined dashes. The latest commit streamlined everything over to the dashed version but I'll not change again. Hopefully this is now finally good to merge. @vdukhovni fyi as I had to touch everything, I also already updated the entropy param to "ikme" so this is in line with your PR (to hopefully also land soon now).